.github/workflows/e2e.yaml

```diff
@@ -0,0 +1,101 @@
+name: E2E
+on:
+  workflow_call:
+    inputs:
+      region:
+        type: string
+        default: "us-east-1"
+      git_repo:
+        type: string
+        default: "aws/karpenter"
+      git_ref:
+        type: string
+      suite:
+        type: string
+        required: true
+      event_name:
+        type: string
+        required: true
+    secrets:
+      SLACK_WEBHOOK_URL:
+        required: true
+env:
+  AWS_REGION: ${{ inputs.region }}
+  K8s_VERSION: 1.25
```

Comment:
Would be nice to parameterize this

Reply:
There isn't a good way to pass these down from their top-level caller. i.e. even if we passed these down as inputs from the top-level workflow caller, we couldn't define these values only once. I tried it and you can't specify environment variables declared once in a `with:` statement like

```yaml
  e2e-upgrade:
    if: github.repository == 'aws/karpenter'
    uses: ./.github/workflows/e2e-upgrade.yaml
    with:
      region: ${{ env.AWS_REGION }} # This fails with an error
      from_git_ref: v0.26.1
      git_repo: jonathan-innis/karpenter
```

Reply:
I also would say that `input` values for wokrflows should be things that are likely to change while environment variables are going to be consistent and generally the same throughout a given environment. I don't see these values as likely to change as often so I think they are fine as environment variables.